src/hotspot/cpu/aarch64/c1_Runtime1_aarch64.cpp

```diff
@@ -443,13 +445,12 @@ OopMapSet* Runtime1::generate_handle_exception(StubID id, StubAssembler *sasm) {
   // r0: handler address
   //      will be the deopt blob if nmethod was deoptimized while we looked up
   //      handler regardless of whether handler existed in the nmethod.
+  //      Move it into LR ready for returning.
+  __ mov(lr, r0);
 
   // only r0 is valid at this time, all other registers have been destroyed by the runtime call
   __ invalidate_registers(false, true, true, true, true, true);
 
-  // patch the return address, this stub will directly return to the exception handler
-  __ str(r0, Address(rfp, 1*BytesPerWord));
-
```

Comment:
Please explain the reason for this change, that leaves `lr` live across `restore_live_registers()`.

Reply:
In the original code:
*save r0 to the lr location on the stack
*restore_live_registers
*Standard return: remove stack frame, load lr and fp off the stack, jump to lr.
 
With PAC it would now be:
*Sign r0 then save it to the lr location on the stack
*restore_live_registers
*Standard return: remove stack frame, load lr and fp off the stack, auth lr, jump to lr.

After reading the code in restore_live_registers, it doesn't touch lr and so seemed odd to have the save to the stack, only to restore it directly afterwards.

Reply:
That's an optimization, though. You shouldn't need to read the code in `restore_live_registers()` to see if it's safe to keep the return address in LR: at best it's pathological coupling, in the sense that the correctness of this code depends on the internal details of  `restore_live_registers()`. Let's keep LR live ranges as short as possible.

Reply:
Ok, that's fine, I'll update it (It'll simplify the total code diff too).